sycl/source/detail/device_filter.cpp

```diff
@@ -124,6 +124,51 @@ void device_filter_list::addFilter(device_filter &Filter) {
   FilterList.push_back(Filter);
 }
 
+// Backend is compatible with the SYCL_DEVICE_FILTER in the following cases.
+// 1. Filter backend is '*' which means ANY backend.
+// 2. Filter backend match exactly with the given 'Backend'
+bool device_filter_list::backendCompatible(backend Backend) {
+  for (const device_filter &Filter : FilterList) {
+    backend FilterBackend = Filter.Backend;
+    if (FilterBackend == Backend || FilterBackend == backend::all)
+      return true;
+  }
+  return false;
+}
+
+bool device_filter_list::deviceTypeCompatible(info::device_type DeviceType) {
+  for (const device_filter &Filter : FilterList) {
+    info::device_type FilterDevType = Filter.DeviceType;
+    if (FilterDevType == DeviceType || FilterDevType == info::device_type::all)
+      return true;
+  }
+  return false;
+}
+
+bool device_filter_list::deviceNumberCompatible(int DeviceNum) {
+  for (const device_filter &Filter : FilterList) {
+    int FilterDevNum = Filter.DeviceNum;
+    if (!Filter.HasDeviceNum || FilterDevNum == DeviceNum)
+      return true;
+  }
+  return false;
+}
+
+bool device_filter_list::containsHost() {
+  for (const device_filter &Filter : FilterList) {
```

Comment:
Could you please clarify logic of check of this function?
It's not clear why it returns `true` if `SYCL_DEVICE_FILTER="opencl:host"` and `false` if SYCL_DEVICE_FILTER="*:*:1".

Reply:
You found a bug.
opencl:host should return false because it is an invalid backend:device_type combination.
device_num==1 should be false because currently we never add HOST device more than once in SYCL RT.